### PR TITLE
Unit Tests for GraspData and GraspGeneratorTest

### PR DESCRIPTION
**Pull request by davetcoleman**
Contains previous PR

### CODE REVIEW

**Comment by mlautman** [comment]
failing because of clang

File: src/demo/grasp_filter_test.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ class GraspGeneratorTest
     // Filter the grasp for only the ones that are reachable
     ROS_INFO_STREAM_NAMED("test", "Filtering grasps kinematically");
     bool filter_pregrasps = true;
-    bool verbose = false;  // note: setting this to true will disable threading
+    // bool verbose = false;  // note: setting this to true will disable threading
```
Comment (mcevoyandy) [inline src/demo/grasp_filter_test.cpp]: `verbose` appears unused... should this just be removed?

File: src/demo/grasp_filter_test.cpp [hunk 2]
```diff
@@ -181,7 +181,7 @@ class GraspGeneratorTest
       // Filter the grasp for only the ones that are reachable
       ROS_INFO_STREAM_NAMED("test", "Filtering grasps kinematically");
       bool filter_pregrasps = true;
-      bool verbose = false;  // note: setting this to true will disable threading
+      // bool verbose = false;  // note: setting this to true will disable threading
```
Comment (mcevoyandy) [inline src/demo/grasp_filter_test.cpp]: same. Remove `verbose`?

**Comment by davetcoleman** [comment]
Fixed clang and andy's feedback

**Pull request merged**